core/internal/runconsolelogs/runconsolelogs.go

```diff
@@ -68,6 +68,10 @@ func New(params Params) *Sender {
 		panic("runconsolelogs: ExtraWork is nil")
 	}
 
+	if params.Settings.GetConsole() == "off" {
+		return nil
+	}
```

Comment:
Conventionally, New() should never return nil unless it returns a (*Value, error) pair.

It would be better to leave all these details to `runconsolelogs.Sender` by checking settings in its methods rather than using a nil value. Specifically, if console capture is off, then `StreamLogs` should be a no-op and `Finish` should skip the file upload. `Finish` should still wait for `s.writer` and `New` should be unmodified.

Reply:
yep, that was my first approach but though maybe this is more obvious can move it inside the console sender